carts/models.py

```diff
@@ -0,0 +1,65 @@
+"""
+Models for cart app
+"""
+from decimal import Decimal
+
+from django.core.validators import MinValueValidator
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from products.models import AuditTimeStamp, Product
+from users.models import User
+
+
+class Cart(AuditTimeStamp):
+    """
+    Cart model
+    """
+
+    objects = models.Manager()
+    user = models.OneToOneField(User, on_delete=models.CASCADE, related_name="cart")
```

Comment:
I think We don't want this relation to be deleted.

Reply:
Arslan bhai, if a user gets deleted should not we remove all the data associated with that user like his cart, cart_items, etc? If not then should it be `on_delete=models.DO_NOTHING` ?

Reply:
We should use `PROTECT` and not let the data be deleted. Additionally, I had one more concern that I somehow failed to mention earlier. A user can have multiple carts e.g. A user adds items to the cart and then submits an order which let's say is delivered but we will still have his cart object but maybe in an inactive state so that a new cart is created when. the user again adds anything into the cart.